lib/ansible/plugins/cliconf/cnos.py

```diff
@@ -76,12 +76,34 @@ def get_config(self, source='running', format='text'):
         return self.send_command(cmd)
 
     @enable_mode
-    def edit_config(self, command):
-        for cmd in chain(['configure terminal'], to_list(command), ['end']):
-            self.send_command(cmd)
+    def edit_config(self, candidate=None, commit=True,
+                    replace=None, comment=None):
+        resp = {}
+        results = []
+        requests = []
+        if commit:
+            self.send_command('configure terminal')
+            for line in to_list(candidate):
+                if not isinstance(line, Mapping):
+                    line = {'command': line}
+
+                cmd = line['command']
+                if cmd != 'end' and cmd[0] != '!':
+                    results.append(self.send_command(**line))
+                    requests.append(cmd)
+
+            self.send_command('end')
+        else:
+            raise ValueError('check mode is not supported')
+
+        resp['request'] = requests
+        resp['response'] = results
+        return resp
 
-    def get(self, command, prompt=None, answer=None, sendonly=False, check_all=False):
-        return self.send_command(command=command, prompt=prompt, answer=answer, sendonly=sendonly, check_all=check_all)
+    def get(self, command, prompt=None, answer=None, sendonly=False,
```

Comment:
There's no change here, right? Just adding newlines?

Reply:
yes, just to make it pep8 compliant

Reply:
Alright, that's fine. I think ansible's current convention is limit only to 120 characters (or maybe longer), but I can't find any references to it. You are, of course, welcome to limit it to whatever you like, as long as it's readable and sanity tests don't complain